lib/templates/lab2/lab2.dart

```diff
@@ -1,19 +1,79 @@
-class Book {
-  // your code here
+
+import 'dart:ffi';
+
+class Book with JsonSerializable {
+  final String title;
+  final String author;
+  final int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+  toJson(); 
+
+  void bookDetails() {
+    print('Title: $title, Author: $author, Number of Pages: $numberOfPages');
+  }
 }
 
-class FictionBook {
-  // your code here
+
+class FictionBook extends Book {
+  String? genre;
+  FictionBook(super.title, super.author, super.numberOfPages, this.genre);
+  void bookDetails() {
+    print('Title: $title, Author: $author, Number of Pages: $numberOfPages, Genre: $genre');
+  }
 }
 
+
 mixin JsonSerializable {
-  // your code here
+  String? title;
+  String? author;
+  int? numberOfPages;
+  Map<String, dynamic> toJson() {
+    return {
+      'title': title,
+      'author': author,
+      'numberOfPages': numberOfPages,
+    };
+  }
 }
 
-double calculateArea(){
-  // your code here
+
+double calculateArea({required double length, required double width, String shape = 'rectangle'}) {
+  if (shape == 'circle') {
+    return 3.14 * length * length;
+  } else {
+    return length * width;
+  }
 }
 
+
 class User{
-  // your code here
+  int? age;
+  String? email;
+  String name;
+  User(this.name, this.age, this.email);
```

Comment:
these params are positional, so you cannot use them in constuctor, so I cant create user with null email

Reply:
And you had to make it required